### ISSUE TITLE
Nginx config is not up to date

### ISSUE THREAD

**Issue by Matzz** [nextcloud/docker#748]
According to 
https://docs.nextcloud.com/server/15/admin_manual/installation/nginx.html
`location ~ ^/(?:index|remote|public|cron|core/ajax/update|status|ocs/v[12]|updater/.+|ocs-provider/.+)\.php(?:$|/) {` should be replaced with `location ~ ^\/(?:index|remote|public|cron|core\/ajax\/update|status|ocs\/v[12]|updater\/.+|oc[ms]-provider\/.+)\.php(?:$|\/) {`
So `ocs` should be replaced with `oc[ms]-`.

I could make a PR to fix that but first of all I would like to know if nginx is officially supported here?
I'm asking because I also noticed that other bits are missing. Eg.
```
    # Remove X-Powered-By, which is an information leak
    fastcgi_hide_header X-Powered-By;
```
I think updating https://docs.nextcloud.com/server/16/admin_manual/installation/nginx.html separately and all nginx.conf files here is impossible. It's just too much copy&paste.
I think there should be one repository which contains common configuration parts which are composable using `include` (https://serverfault.com/questions/707955/nginx-split-large-configuration-file). What do you think about that? I could try to do it but I would need some help. First of all I'm not relay sure in which repo I should create those files and how to reference them in `nextcloud/docker` repository? What's more I'm not an nginx specialist so I would like to ask for volunteer to review my changes.

cc @tilosp as you created most of those files

**Comment by J0WI** [comment]
We provide some [example](https://github.com/nextcloud/docker/tree/master/.examples) configurations, but they are not available as official images.
Please refer to https://github.com/nextcloud/documentation for enhancements in the documentation itself.

**Comment by onzo-mateuszzakarczemny** [comment]
@J0WI So should I close this ticket?

**Comment by Matzz** [comment]
@JOWI So should I close this ticket?

**Comment by J0WI** [comment]
I'd welcome a PR to update the `ocs` regex.

**Comment by J0WI** [comment]
Closed by #912
